### PR TITLE
[cmake] addons: windows: only add sse2 flags for x86

### PR DESCRIPTION
**Pull request by lrusak**
I noticed that the build log for win64 has a lot of lines like this:
```
20:18:35 [  2%] Building CXX object CMakeFiles/peripheral.joystick.dir/src/addon.cpp.obj
20:18:35 cl : Command line warning D9002 : ignoring unknown option '/arch:SSE2'
``` 
This should clear that up

ref: https://stackoverflow.com/questions/1067630/sse2-option-in-visual-c-x64

## What is the effect on users?

less spam in the build log?

### CODE REVIEW

**Comment by thexai** [comment]
Fine but commit/PR title seems confusing:

You cannot "remove SSE2 support" from x64 because SSE2 instructions are supported on all x64 processors. Then maybe something like:

"Remove /SSE2 build flag from x64 as SSE2 support is already enabled by default"

or 

"remove unnecessary /SSE2 build flag from x64"

File: cmake/scripts/windows/CXXFlagOverrides.cmake [hunk 1]
```diff
@@ -4,7 +4,11 @@ if(MSVC)
...
     set(MP_FLAG "/MP")
   endif()
-  set(CMAKE_CXX_FLAGS "/D_UNICODE /DUNICODE /DRPC_USE_NATIVE_WCHAR ${MP_FLAG} /DWIN32 /D_WINDOWS /W3 /GR /Zi /EHsc /arch:SSE2")
+  set(CMAKE_CXX_FLAGS "/D_UNICODE /DUNICODE /DRPC_USE_NATIVE_WCHAR ${MP_FLAG} /DWIN32 /D_WINDOWS /W3 /GR /Zi /EHsc")
   set(CMAKE_CXX_FLAGS_DEBUG "/D_DEBUG /MDd /Ob0 /Od /RTC1 /D_ITERATOR_DEBUG_LEVEL=0")
   set(CMAKE_CXX_FLAGS_RELEASE "/MD /Ox /Ob2 /Oi /Ot /Oy /GL /DNDEBUG")
 endif()
+
+if(NOT "$ENV{Platform}" STREQUAL "x64")
+  string(JOIN " " CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}" "/arch:SSE2")
```
Comment (thexai) [inline cmake/scripts/windows/CXXFlagOverrides.cmake:13]: Maybe is more simple:

`string(APPEND CMAKE_CXX_FLAGS " /arch:SSE2")`

**Comment by wsnipex** [comment]
I guess https://github.com/xbmc/xbmc/blob/master/cmake/modules/FindSSE.cmake#L130-L137 is redundant

**Comment by lrusak** [comment]
So I think there is some misunderstanding here as this only effects the add-on build IIRC. These scripts only have some variables available.

```
message("TARGET_ARCHITECTURE: ${TARGET_ARCHITECTURE}")
message("cmakeArch: ${cmakeArch}")
message("Platform: $ENV{Platform}")
message("ARCH: ${ARCH}")
message("CMAKE_GENERATOR_PLATFORM: ${CMAKE_GENERATOR_PLATFORM}")
```

x64
```
14:46:08 TARGET_ARCHITECTURE: 
14:46:08 cmakeArch: 
14:46:08 Platform: x64
14:46:08 ARCH: 
14:46:08 CMAKE_GENERATOR_PLATFORM: 
```

x86
```
14:49:27 TARGET_ARCHITECTURE: 
14:49:27 cmakeArch: 
14:49:27 Platform: x86
14:49:27 ARCH: 
14:49:27 CMAKE_GENERATOR_PLATFORM: 
```

**Review by thexai — APPROVED** [review]
The changes are correct and although variable value is `x86` in commit name it would be more correct to talk about `win32` because x86 is both 32 and 64 bits

**Comment by lrusak** [comment]
> The changes are correct and although variable value is `x86` in commit name it would be more correct to talk about `win32` because x86 is both 32 and 64 bits

well it seems that windows refers to `x86` as 32bit and `x64` as 64bit no?

**Review by AlwinEsch — APPROVED** [review]
With more less warnings is always good, thanks about the fix!

**Pull request merged**